internal/common/metrics.go

```diff
@@ -37,8 +33,6 @@ func StartSendingMetrics(receivers []string, agentRunDuration time.Duration, dat
 			multiErr = multierr.Append(multiErr, err)
 		}(receiver, agentRunDuration)
 	}
-
-	wg.Wait()
```

Comment:
I think the wait group is still needed. Otherwise the returned `error` won't be accurate. The function returns before any `multiErr` variable can get updated.

Reply:
Forgot to delete the go function. Therefore, we don't need to use wg here though since every sending metrics inside the metrics are a ticker.